### PR TITLE
4 3 make domain manage

### PR DESCRIPTION
**Pull request by tomato1976**

### CODE REVIEW

File: app/controllers/user_domains_controller.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  rescue 
+    flash[:error] = "There is the same Domain [" + @domain.domain + "] "
+    redirect_to new_user_domain_path
+  end
+
+  def destroy
+    @domain = Domain.find(params[:id])
+    if session[:search_scope] && (session[:search_scope]["domain_name"] == @domain.domain)
+      # session.delete(:search_scope)
+      session[:search_scope]=nil
```
Comment (tomato1976) [inline app/controllers/user_domains_controller.rb]: Line32.
Is this a right way to delete session by setting nil to my session?

Line31. can't work....

Comment (rizwanreza) [inline app/controllers/user_domains_controller.rb]: Try this: `session.delete(:search_scope)`

File: app/views/site_logs/index.html.haml [hunk 2]
```diff
@@ -1,115 +1,117 @@
...
-        }
-    ]
-  };
-
-  // draw chart
-  var myBarChartYear  = new Chart(document.getElementById("yearChart").getContext("2d")).Bar(dataYear);
-  var myBarChartMonth  = new Chart(document.getElementById("monthChart").getContext("2d")).Bar(dataMonth);
-  var myBarChartDay  = new Chart(document.getElementById("dayChart").getContext("2d")).Bar(dataDay);
 
+-if current_user
```
Comment (rizwanreza) [inline app/views/site_logs/index.html.haml]: Try using spaces after `-`, like this: `- if current_user`

File: app/views/user_domains/edit.html.haml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+%h1 UserDomains#edit
+%p Find me in app/views/user_domains/edit.html.haml
```
Comment (rizwanreza) [inline app/views/user_domains/edit.html.haml]: Maybe remove this file?

File: app/controllers/site_logs_controller.rb [hunk 4]
```diff
@@ -1,62 +1,46 @@
...
-    for hour in "00".."23" do
-       @specified_day_hours_pv.store(hour,  hours_pv.keys.include?(hour) ? hours_pv[hour] : 0 )
+  def index
+    if current_user
+        # todo: we have to change
+      if params[:search_scope]
+        @search_scope = SearchScope.new(params[:search_scope], current_user)
+      else session[:search_scope]
+        @search_scope = SearchScope.new(session[:search_scope], current_user)
+      end
```
Comment (rizwanreza) [inline app/controllers/site_logs_controller.rb]: `@search_scope = SearchScope.new(params[:search_scope] || session[:search_scope], current_user)`

File: app/controllers/user_domains_controller.rb (hunk 1 above)
Comment (rizwanreza) [inline app/controllers/user_domains_controller.rb]: `session.delete(:search_scope); nil`

File: app/models/search_scope.rb [hunk 5]
```diff
@@ -0,0 +1,134 @@
+class SearchScope
```
Comment (rizwanreza) [inline app/models/search_scope.rb]: Add model specs for this file.

File: app/controllers/user_domains_controller.rb (hunk 1 above)
Comment (tomato1976) [inline app/controllers/user_domains_controller.rb]: It worked!
 session.delete(:search_scope)

File: app/views/site_logs/index.html.haml (hunk 2 above)
Comment (tomato1976) [inline app/views/site_logs/index.html.haml]: I changed like that! Thanks.

File: app/views/user_domains/edit.html.haml (hunk 3 above)
Comment (tomato1976) [inline app/views/user_domains/edit.html.haml]: Deleted this file!

File: app/models/search_scope.rb (hunk 5 above)
Comment (tomato1976) [inline app/models/search_scope.rb]: OK!

**Pull request merged**